File: 0121-best_time_to_buy_and_sell_stock.go
func maxProfit(prices []int) int {
	maxProfit := 0
	minPrice := prices[0]

	for _, price := range prices {
		if price < minPrice {
			minPrice = price
		}

		if profit := price - minPrice; profit > maxProfit {
			maxProfit = profit
		}
	}

	return maxProfit
}

File: 0121-best_time_to_buy_and_sell_stock.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        sell_day = 1
        buy_day = 0

        while(sell_day < len(prices)):
          if (prices[sell_day] > prices[buy_day]):
            profit = prices[sell_day] - prices[buy_day]

            max_profit = max(max_profit, profit)
          else:
            buy_day = sell_day

          sell_day += 1

        return max_profit

            